spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/QueryUtils.java

```diff
@@ -165,4 +169,23 @@ static Delete createDeleteQuery(String tableName, Object objectToDelete, QueryOp
 
 		return new SliceImpl<>(result, pageRequest, pagingState != null);
 	}
+
+	/**
+	 * Read a {@link Mono<Slice>} of data from the {@link ReactiveResultSet} for a {@link Pageable}.
+	 *
+	 * @param resultSet must not be {@literal null}.
+	 * @param fetchSize must not be {@literal null}.
+	 * @param rowMapper must not be {@literal null}.
+	 * @return the resulting {@link Slice}.
+	 */
+	static <T> Mono<Slice<T>> readSlice(ReactiveResultSet resultSet, Integer fetchSize, Function<Row, T> rowMapper) {
```

Comment:
We can't introduce dependencies to reactive types here as we want the module to run without reactive dependencies as well.

We also need to be careful about fetching/reading progress. If the fetch size differs from the actual number of rows returned for a page, we might request the next page without intending to do so. I think we need another method on `ReactiveResultSet` that gives us a `Flux` without transparent next page fetching.